### ISSUE TITLE
Link to namespace in TAGFILES is not resolved with EXTRACT_ALL=NO

### ISSUE THREAD

**Issue by st-masa** [doxygen/doxygen#9602]
**Describe the bug**

Assuming the document(doxy2) refers symbols in `TAGFILES` generated from the other document(doxy1).
The root namespace 'ns' that is described in doxy1 only cannot be resolved.
Both documents have `EXTRACT_ALL=NO` configuration.

doxygen says the following warnings:

```
extractAllSample/doxy2/sample.dox:4: error: unable to resolve reference to 'ns' for \ref command
extractAllSample/src/header2.h:18: error: unable to resolve reference to 'ns' for \ref command
extractAllSample/doxy2/mainpage.dox:4: error: unable to resolve reference to 'ns' for \ref command
extractAllSample/doxy2/mainpage.dox:4: error: unable to resolve reference to 'ns' for \ref command
```

Here are the contents of documents:

* doxy1
  * namespace
    * ns, ns::sample1
  * files
    * ns.h, header1.h
  * output
    * doxygen_docs/html1
* doxy2
  * namespace
    * ns::sample2
  * files
    * header2.h, mainpage.dox, sample.dox
  * output
    * doxygen_docs/html2

Summarize of observed behavior

* EXTRACT_ALL configuration in doxy2 (version 1.9.5)
  * doxy2 with `EXTRACT_ALL=NO`
    * Cannot resolve reference to `ns`.
  * doxy2 with `EXTRACT_ALL=YES`
    * Can resolve reference to `ns`.

* doxygen version (with `EXTRACT_ALL=NO`)
  * 1.9.3
    * doxy2 contains namespace `ns`. 
  * 1.9.5 and 1.9.4
    * doxy2 does not contain namespace `ns` and `ns::sample1`.
    * `ns::sample1` is linked from ns::sample2 page. (namespacens_1_1sample2.html)
    * `ns` is not linked from ns::sample2 page. (namespacens_1_1sample2.html)

* My doubts about these behaviors
  * `EXTRACT_ALL=NO` does not allow reference links from `TAGFILES` (treated as undocumented)?
    * If so, it seems strange that `ns::sample1` is linked in doxy2.
  * 1.9.5's doxy2 does not document `ns`. Is this behavior intended?

**Expected behavior**
* namespaces derived from `TAGFILES` are successfully linked.
* namespaces derived from `TAGFILES` are listed in "Namespace List".

**Screenshots**

* 1.9.5 `ns` is not linked
![nspaces_not_linked](https://user-images.githubusercontent.com/79039413/190160605-6cdd6331-0b33-4b52-903f-0aad8ce94423.png)

* 1.9.5 No namespace is listed in doxy2 document
![no_nspace_listed](https://user-images.githubusercontent.com/79039413/190160651-50dd1559-6cb1-4829-b302-3d8eb484d9e9.png)

* 1.9.3 All namespaces are listed in doxy2 document
![all_nspaces_listed](https://user-images.githubusercontent.com/79039413/190160682-e864246d-bdbe-4312-a86b-0636c29bcf52.png)

**To Reproduce**

1. Extract the attachment [extractAllSample.zip](https://github.com/doxygen/doxygen/files/9566484/extractAllSample.zip).
2. Create doxy1 document
```
cd ./doxy1
doxygen ./DoxyFile
```
3. Create doxy2 document
```
cd ../doxy2
doxygen ./DoxyFile
```
4. Browse doxygen_docs/html2 to check namespace links.

**Version**
1.9.5 on Windows 10 (64 bit) and MacOSX.
Seems to exist in older versions.

**Comment by doxygen** [comment]
I had a first look at the problem.

The bare use of `ns` will never result in a link as it seen as a normal word. So you need to change it into `::ns` or `#ns` to make doxygen look for a link.

The `@ref ns` (or `::ns`) link in the example still doesn't work, because even though the namespace is listed in the tag file, it is also found as an undocumented namespace in the doxy2 project. As a result doxygen concludes it cannot link to the namespace. I'll see if this can be improved.

**Comment by doxygen** [comment]
@st-masa Please verify that the referenced commit fixes the linking issue for you (but check the comment above as well).

**Comment by st-masa** [comment]
@doxygen Thank you for the fix.
I have confirmed the following fixes with 1.9.6 (68fdd72c6047723d0457106dda3010e13e4a5fa9):

* doxy2 with EXTRACT_ALL=NO
  * Can resolve reference to ns with @ref cmd, `::ns` and `#ns`.
* doxy2 does contain namespace `ns` and `ns::sample1`.
* `ns` with @ref cmd is linked from ns::sample2 page. (namespacens_1_1sample2.html)

And I found some suspicious behaviors. So I have added some information to the sample.
Please see the attachment [extractAllSample2.zip](https://github.com/doxygen/doxygen/files/9646625/extractAllSample2.zip).

* Added function NsFunction*() under namespace `ns` in each header so that the function which belongs to `ns` is documented.
* Added class NsClass* under namespace `ns` in each header so that the class which belongs to `ns` is documented.

* In header2.h, added "#if " line to switch documentation state of `ns::sample2`. Its behavior is different depending on this line.
  * `#if 0` - `ns::sample2` is undocumented.
    * namespaces.html is not created.
    * `NsClass1` (derived from doxy1) is not found in doxy2 Class List.
  * `#if 1` - `ns::sample2` is documented.
    * `NsFunction2` is not documented inside namespace `ns` in doxy2.
    * `NsFunction2` is listed in Namespace List. But that is linked to doxy1's `ns`, not linked correctly.
    * `NsFunction2` is listed in header2.h File List. But that is linked to doxy1's `ns`, not linked correctly.

**Comment by st-masa** [comment]
> should be in sample2?
Sorry, this is because class definitions are not terminated by semicolon.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
